source/utilities.cc

```diff
@@ -1724,11 +1724,103 @@ namespace aspect
+                                 const std::vector<Table<dim,double> > &raw_data)
+    {
+      Assert(coordinate_values_.size()==dim, ExcMessage("Invalid size of coordinate_values."));
+      for (unsigned int d=0; d<dim; ++d)
+        {
+          coordinate_values[d] = coordinate_values_[d];
+          AssertThrow(coordinate_values[d].size()>1,
+                      ExcMessage("Error: At least 2 entries per coordinate direction are required."));
+          table_points[d] = coordinate_values_[d].size();
+        }
+
+      components = column_names.size();
+      data_component_names = column_names;
+      Assert(raw_data.size() == components,
+             ExcMessage("Error: Incorrect number of columns specified."));
+
+      // compute maximum_component_value for each component:
+      maximum_component_value = std::vector<double>(components,-std::numeric_limits<double>::max());
+      for (unsigned int c=0; c<components; ++c)
+        {
+          Assert(raw_data[c].size() == table_points,
+                 ExcMessage("Error: One of the data tables has an incorrect size."));
+
+          const unsigned int n_elements = raw_data[c].n_elements();
+          for (unsigned int idx=0; idx<n_elements; ++idx)
+            maximum_component_value[c] = std::max(maximum_component_value[c], raw_data[c](
+                                                    compute_table_indices(table_points, idx)));
+        }
+
+      // In case the data is specified on a grid that is equidistant
+      // in each coordinate direction, we only need to store
+      // (besides the data) the number of intervals in each direction and
+      // the begin- and endpoints of the coordinates.
+      // In case the grid is not equidistant, we need to keep
+      // all the coordinates in each direction, which is more costly.
+      std::array<unsigned int,dim> table_intervals;
+
+      coordinate_values_are_equidistant = true;
+      for (unsigned int d=0; d<dim; ++d)
+        {
+          table_intervals[d] = table_points[d]-1;
+
+          // The minimum and maximum coordinate values:
+          grid_extent[d].first = coordinate_values[d][0];
+          grid_extent[d].second = coordinate_values[d][table_points[d]-1];
+
+          const double grid_spacing = coordinate_values[d][1] - coordinate_values[d][0];
+
+          for (unsigned int n = 1; n < table_points[d]; n++)
+            {
+              const double current_grid_spacing = coordinate_values[d][n] - coordinate_values_[d][n-1];
+
+              AssertThrow(current_grid_spacing > 0.0,
+                          ExcMessage ("Coordinates in dimension "
+                                      + Utilities::int_to_string(d)
+                                      + " are not strictly ascending. "));
+
+              if (std::abs(current_grid_spacing - grid_spacing) > 0.005*(current_grid_spacing+grid_spacing))
+                coordinate_values_are_equidistant = false;
```

Comment:
You can `break` here -- it's never going back to `true` :-)

Reply:
Yes, but I want to run through and do the AssertThrow check.

Reply:
Ah, I see. Can you add a comment here then why you're not `break`ing right away?